### PR TITLE
[ESIMD] Allow GenX intrinsics to pass LLVM-SPRIV translation

### PR DESCRIPTION
**Pull request by DenisBakhvalov**
This is one of the final patches to support mixing SYCL and ESIMD kernels.
To make this happen, we need to allow GenX intrinsics to pass LLVM-SPRIV translation.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/SYCL.cpp [hunk 1]
```diff
@@ -43,8 +43,7 @@ const char *SYCL::Linker::constructLLVMSpirvCommand(
     CmdArgs.push_back("-spirv-ext=+all");
     CmdArgs.push_back("-spirv-debug-info-version=legacy");
     CmdArgs.push_back("-spirv-allow-extra-diexpressions");
-    if (C.getArgs().hasArg(options::OPT_fsycl_esimd))
-      CmdArgs.push_back("-spirv-allow-unknown-intrinsics");
+    CmdArgs.push_back("-spirv-allow-unknown-intrinsics=llvm.genx.");
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/SYCL.cpp:46]: Definitely not very efficient having to make 2 changes.  This needs to be refactored (not by you, but an issue should be created)

Comment (DenisBakhvalov) [reply clang/lib/Driver/ToolChains/SYCL.cpp:46]: Opened https://github.com/intel/llvm/issues/3451.

**Review by mdtoguchi — APPROVED** [review]

**Review by kbobrovs — APPROVED** [review]
LGTM

**Comment by DenisBakhvalov** [comment]
@AGindinson, please review,

**Review by AGindinson — COMMENTED** [review]
Obviously, I'm late to the party - LGTM for the driver changes nonetheless. Do I get it that more ESIMD-related changes in other components are needed to enable end-to-end ESIMD + SYCL testing?

**Comment by DenisBakhvalov** [comment]
> Do I get it that more ESIMD-related changes in other components are needed to enable end-to-end ESIMD + SYCL testing?

Actually, it is already enabled and a simple test is already working.
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/sycl_esimd_mix.cpp

**Pull request merged**